redfish-core/lib/assembly.hpp

```diff
@@ -365,9 +421,87 @@ inline void setAssemblylocationIndicators(
         {
             setLocationIndicatorActive(asyncResp, assembly, iter->second);
         }
+  
+        // Handle special case for tod_battery assembly OEM ReadyToRemove property
+        // NOTE: The following method for the special case of the tod_battery
+        // ReadyToRemove property only works when there is only ONE adcsensor
+        // handled by the adcsensor application.
+        if (sdbusplus::message::object_path(assembly).filename() == "tod_battery")
+        {
+            if (oem)
```

Comment:
Isn't this just the last oem from the iteration above?

Reply:
The above FOR loop at Line: 415 "for (const auto& assembly : assemblies)" is looping through the ENTIRE LIST of 30+ assemblies.  When it gets to the one with the associated name "tod_battery" it will check if "oem" is true, and it will be if it was included in the PATCH request.  So ALL assemblies are checked until the battery assembly is found and then it is checked for oem true.  In addition, the above loop is also setting all the LEDs according to and LocationIndicatorActive LED flags that were supplied in the PATCH request. (That was originaly there.)

Reply:
Above you stated we must be able to process the following commands independently of the other; which we can now do.

1) curl -k -X PATCH -d '{"Assemblies":[{"MemberId" : "1", "LocationIndicatorActive": true]}' https://root:0penBmc@ever8bmc:18080/redfish/v1/Chassis/chassis/Assembly

2) curl -k -X PATCH -d '{"Assemblies":[{"Oem": { "OpenBMC": { "ReadyToRemove": true }}}]}' https://root:0penBmc@ever8bmc:18080/redfish/v1/Chassis/chassis/Assembly

If you look at command #2 we have ["Oem": { "OpenBMC": { "ReadyToRemove"].  There is nothing there to identify that with any particular assembly of the 30+ that we have.  We are looking for the battery assembly and applying it to that.  IF, in the future, some other assembly has an 
"Oem": { "OpenBMC": { "ReadyToRemove" property on it, then how are we going to know which assembly the incoming "Oem": { "OpenBMC": { "ReadyToRemove" PATCH is for?  You indicated that we can not use memberID because it may change.
(Note that the LEDs are using memberId and that was the original design prior to the battery work)

We either have a DESIGN PROBLEM or we must assume there is only one assembly with an 
"Oem": { "OpenBMC": { "ReadyToRemove" property on it and that it goes to the battery assembly.